frontend/src/metabase/visualizer/visualizations/cartesian.ts

```diff
@@ -25,13 +26,13 @@ import type {
   VisualizerColumnReference,
   VisualizerDataSource,
   VisualizerDataSourceId,
-  VisualizerHistoryItem,
+  VisualizerVizState,
 } from "metabase-types/store/visualizer";
 
 import { removeColumnFromStateUnlessUsedElseWhere } from "./utils";
 
 export const cartesianDropHandler = (
-  state: VisualizerHistoryItem,
+  state: Draft<VisualizerVizState> | VisualizerVizState,
```

Comment:
This "Type instantiation is excessively deep" is sort of a whack-a-mole. I thought I got rid of it after removing `maybeUpdateHistory`, but TSC started erroring elsewhere. Adding `Draft` type + a union with the original type seems to solve it. We have a similar fix in the [dashboard code](https://github.com/metabase/metabase/blob/6bad11bfeb8b9a29bb2d1435d5373cbcf160acbe/frontend/src/metabase/dashboard/actions/tabs.ts#L109)